Ext4Pkg/Ext4Dxe/Ext4Dxe.h

```diff
@@ -379,6 +383,8 @@ struct _Ext4File {
   EXT4_DENTRY           *Dentry;
 };
 
+#define EXT4_FILE_FROM_THIS(This)  BASE_CR ((This), EXT4_FILE, Protocol)

```

Comment:
It's EDK II convention to use `CR` and have a signature to detect basic memory corruption. Maybe comply?

Reply:
I agree it is reliable. Done in https://github.com/acidanthera/audk/pull/11/commits/a9a0d967cbf882d652139e928761579d3c07b25b